airbyte-integrations/connectors/source-amazon-ads/unit_tests/test_report_streams.py

```diff
@@ -612,3 +609,37 @@ def test_streams_state_filter(mocker, config, state_filter, stream_class):
         assert params["stateFilter"] == ",".join(state_filter)
     else:
         assert state_filter is None
+
+
+@responses.activate
+def test_display_report_stream_single_record_type(config_gen):
+    setup_responses(
+        init_response=REPORT_INIT_RESPONSE,
+        status_response=REPORT_STATUS_RESPONSE,
+        metric_response=METRIC_RESPONSE,
+    )
+
+    profiles = make_profiles()
+    custom_record_types = ["campaigns"]
+
+    stream = SponsoredDisplayReportStream(config_gen(report_record_types=custom_record_types), profiles, authenticator=mock.MagicMock())
+    stream_slice = {"profile": profiles[0], "reportDate": "20210725"}
+    metrics = [m for m in stream.read_records(SyncMode.incremental, stream_slice=stream_slice)]
+    assert len(metrics) == len(custom_record_types) * len(stream.metrics_map)
```

Comment:
Assertion for the length doesn't ensure that the retrieved record types are the same as requested. Instead, assert whether the set of retrieved `record["recordType"]` is equal to the set of `custom_record_types`.

Reply:
makes sense